Core/include/Acts/Navigation/MultiWireLayerUpdators.hpp

```diff
@@ -0,0 +1,199 @@
+      }
+
+      // Fill with the indexed surfaces and the portals of the volume
+
+      const auto m_indexedSurfaces = getIndexedSurfacesFromVolume(nState);
+
+      auto tposition = nState.position;
+      auto tdirection = nState.direction;
+
+      const auto& grid = m_indexedSurfaces.grid;
+      const auto& extractor = m_indexedSurfaces.extractor;
+      auto nBins = grid.numLocalBins();
+      auto binWidth = grid.binWidth();
+      auto minPos = grid.minPosition();
+      auto maxPos = grid.maxPosition();
+
+      Vector3 origin = Vector3(0, minPos[1], 0);
+      Translation3 trans(origin);
+      Transform3 transform(trans);
+
+      auto dy = binWidth[1];
+      auto phi = Acts::VectorHelpers::phi(tdirection);
+
+      BinUtility binUtility2D(nBins[0], minPos[0], maxPos[0], open, binX,
+                              transform);
+      binUtility2D +=
+          BinUtility(nBins[1], minPos[1], maxPos[1], open, binY, transform);
+
+      for (std::size_t i = 0; i < nBins[1]; i++) {
+        // auto lbin = grid.localBinsFromPosition(tposition);
+        // auto lbin = grid.globalBinFromPosition(tposition);
+        Vector2 localposition =
+            Vector2(tposition.x() + tolerance, tposition.y() + tolerance);
+        /* std::cout<<"local
+         position="<<localposition.x()<<","<<localposition.y()<<std::endl;
+         std::cout<<"bin utility bin="<<binUtility2D.bin(tposition)<<std::endl;
+         std::cout<<"bin utility
+         bin="<<binUtility2D.bin(tposition,1)<<std::endl;*/
+
+        std::array<size_t, 2> lbin = {binUtility2D.bin(localposition) + 1,
+                                      binUtility2D.bin(localposition, 1) + 1};
+        auto extracted =
+            extractor.extract(gctx, nState, grid.atLocalBins(lbin));
+        // auto extracted = extractor.extract(gctx, nState,
+        // grid.atPosition(tposition));
+        std::cout << "lbin=" << lbin[0] << "," << lbin[1] << std::endl;
+        std::cout << "extracted size=" << extracted.size() << std::endl;
+        std::cout << "extracted centers:" << std::endl;
+        std::cout << extracted[0]->center(gctx).y() << "\t";
+        std::cout << extracted[1]->center(gctx).y() << "\t";
+        std::cout << extracted[2]->center(gctx).y() << "\t";
+
+        Acts::Experimental::SurfacesFiller::fill(nState, extracted);
+
+        tposition.x() = tposition.x() + dy / tan(phi);
+        tposition.y() = tposition.y() + dy;
+        /*std::cout<<"in the MultiWireLayer"<<std::endl;
+        std::cout<<nState.surfaceCandidates.size()<<std::endl;
+        std::cout<<"is inside="<<binUtility2D.inside(localposition)<<std::endl;
+        std::cin.ignore();*/
+      }
+
+      const auto& portals = nState.currentVolume->portals();
+      Acts::Experimental::PortalsFiller::fill(nState, portals);
+    }
+
+    updateCandidates(gctx, nState);
+  }
+
+ private:
+  float tolerance = 0.5;
+};
+
+inline static Acts::Experimental::SurfaceCandidatesUpdator MultiWireLayer() {
```

Comment:
I think I would put `SurfaceUpdator` or something else in the name of this function. The name, as is, is a bit misleading. Same for `MultiWireLayerImpl`